src/components/RelatedTitleInfo/RelatedTitleInfo.js

```diff
@@ -14,64 +14,61 @@ import EResourceIdentifier from '../EResourceIdentifier';
 import css from './RelatedTitle.css';
 
 const propTypes = {
-  relatedTitles: PropTypes.arrayOf(PropTypes.shape({
-    titleInstance: PropTypes.shape({
-      id: PropTypes.string,
-      identifiers: PropTypes.arrayOf(PropTypes.shape({
-        identifier: PropTypes.shape({
-          ns: PropTypes.shape({
-            value: PropTypes.string,
-          }),
+  index: PropTypes.number,
+  relatedTitle: PropTypes.shape({
+    id: PropTypes.string,
+    identifiers: PropTypes.arrayOf(PropTypes.shape({
+      identifier: PropTypes.shape({
+        ns: PropTypes.shape({
           value: PropTypes.string,
         }),
-      })),
-      longName: PropTypes.string,
-      name: PropTypes.string,
-      subType: PropTypes.shape({
-        id: PropTypes.string,
-        label: PropTypes.string,
-        value: PropTypes.string,
-      }),
-      type: PropTypes.shape({
-        id: PropTypes.string,
-        label: PropTypes.string,
         value: PropTypes.string,
       }),
+    })),
+    longName: PropTypes.string,
+    name: PropTypes.string,
+    subType: PropTypes.shape({
+      id: PropTypes.string,
+      label: PropTypes.string,
+      value: PropTypes.string,
+    }),
+    type: PropTypes.shape({
+      id: PropTypes.string,
+      label: PropTypes.string,
+      value: PropTypes.string,
     }),
-  })).isRequired,
+  }).isRequired,
 };
 
 const RelatedTitleInfo = ({
-  relatedTitles
+  relatedTitle, index
 }) => (
-  relatedTitles.map((titleInstance, i) => (
-    <div key={`related-title-${i}`} data-test-related-titles>
-      <div className={css.separator} />
-      <Headline data-test-title-instance-name size="large" tag="h3">
-        <FormattedMessage id="ui-agreements.eresources.relatedTitle" values={{ name: titleInstance.name }} />
-      </Headline>
-      <Row>
-        <Col xs={3}>
-          <KeyValue label={<FormattedMessage id="ui-agreements.eresources.materialType" />}>
-            <div data-test-title-instance-sub-type>{titleInstance.subType?.label ?? <NoValue />}</div>
-          </KeyValue>
-        </Col>
-        {titleInstance.type.value === resourceTypes.MONOGRAPH || titleInstance.type.value === resourceTypes.BOOK ?
-          <>
-            <EResourceIdentifier titleInstance={titleInstance} type="isbn" />
-            <EResourceIdentifier titleInstance={titleInstance} type="doi" />
-          </>
-          :
-          <>
-            <EResourceIdentifier titleInstance={titleInstance} type="ezb" />
-            <EResourceIdentifier titleInstance={titleInstance} type="zdb" />
-            <EResourceIdentifier titleInstance={titleInstance} type="eissn" />
-            <EResourceIdentifier titleInstance={titleInstance} type="issn" />
-          </>
+  <div key={`related-title-${index}`} data-test-related-titles>
```

Comment:
Setting the `key` isn't necessary here any more since you're setting it on the component itself. As a result, it appears that you no longer need to pass in `index` as a prop either.